catalog/spatial/registry/registry-publication-update-handler/src/main/java/org/codice/ddf/registry/publication/RegistryPublicationHandler.java

```diff
@@ -0,0 +1,108 @@
+
+import org.apache.commons.collections.CollectionUtils;
+import org.codice.ddf.registry.common.RegistryConstants;
+import org.codice.ddf.registry.common.metacard.RegistryObjectMetacardType;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+public class RegistryPublicationHandler implements EventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryPublicationHandler.class);
+
+    private static final String METACARD_PROPERTY = "ddf.catalog.event.metacard";
+
+    private FederationAdminService adminService;
+
+    private ExecutorService executor;
+
+    private int threadPoolSize = 1;
+
+    public RegistryPublicationHandler(FederationAdminService adminService) {
+        this.adminService = adminService;
+    }
+
+    public void init() {
+        if (executor == null) {
+            executor = Executors.newFixedThreadPool(threadPoolSize);
+        }
+    }
+
+    public void setExecutorService(ExecutorService service) {
+        executor = service;
+    }
+
+    public void destroy() {
+        executor.shutdown();
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        Metacard mcard = (Metacard) event.getProperty(METACARD_PROPERTY);
+        if (mcard == null) {
+            return;
+        }
+        if (mcard.getTags()
+                .contains(RegistryConstants.REGISTRY_TAG)) {
+            executor.execute(() -> {
+                processUpdate(mcard);
+            });
+        }
+    }
+
+    protected void processUpdate(Metacard mcard) {
+        Attribute publishedLocations =
+                mcard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+        if (publishedLocations != null) {
+            Attribute lastPublished = mcard.getAttribute(RegistryObjectMetacardType.LAST_PUBLISHED);
+            Date datePublished = null;
+            if (lastPublished != null) {
+                datePublished = (Date) lastPublished.getValue();
+            }
+
+            Set<String> locations = new HashSet<>(publishedLocations.getValues()).stream()
+                    .map(e -> e.toString())
+                    .collect(Collectors.toSet());
+
+            if ((datePublished == null || datePublished.before(mcard.getModifiedDate()))
+                    && CollectionUtils.isNotEmpty(locations)) {
+                try {
+                    adminService.updateRegistryEntry(mcard, locations);
+                    mcard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.LAST_PUBLISHED,
+                            mcard.getModifiedDate()));
+                    adminService.updateRegistryEntry(mcard);
```

Comment:
The first call is updating a remote registry. If that succeeds then we update our local entry to indicate that. The general framework that is used in ddf doesn't have a mechanism as far as i know to prevent clobbering of changes when updating. That said the chance of us actually doing that here is very small. There is an framework ingest plugin for registry that checks the existing modified date against the incoming metacard modified data and rejects the update if the incoming modified data is older that the one in the catalog.